### PR TITLE
Bug 1148433 - Maintains state of home view in multiple tabs

### PR DESCRIPTION
**Pull request by bkmunar**
Originally tried working on URL changes (anchoring stuff), but it would mess up the back and forward states.
Bug 1170726 --> this is the bug where back/fwd history gets messed up when we go to a page with an anchor on it
Ended up storing a number in Browser.swift called previousHomePanel and updated using a custom function on BrowserViewController.swift when user taps on the home page. 

Filed follow-up bug: Bug 1170769

### CODE REVIEW

File: Client/Frontend/Browser/BrowserViewController.swift [hunk 1]
```diff
@@ -331,7 +331,7 @@ class BrowserViewController: UIViewController {
 
     private func showHomePanelController(#inline: Bool) {
         homePanelIsInline = inline
-
+
```
Comment (thebnich) [inline Client/Frontend/Browser/BrowserViewController.swift]: Please undo this change. You'll want to enable both "Automatically trim trailing whitespace" and "Including whitespace-only lines" preferences in Xcode (under Text Editing)

File: Client/Frontend/Browser/BrowserViewController.swift [hunk 2]
```diff
@@ -342,7 +342,10 @@ class BrowserViewController: UIViewController {
 
             addChildViewController(homePanelController!)
         }
-
+        
+        //updates previous home panel
```
Comment (thebnich) [inline Client/Frontend/Browser/BrowserViewController.swift]: You can probably drop this comment (I think the code below is pretty self-descriptive).

File: Client/Frontend/Browser/BrowserViewController.swift [hunk 3]
```diff
@@ -383,7 +386,7 @@ class BrowserViewController: UIViewController {
 
     private func updateInContentHomePanel(url: NSURL?) {
         if !urlBar.isEditing {
-            if url == AppConstants.AboutHomeURL {
+            if (url == AppConstants.AboutHomeURL) {
```
Comment (thebnich) [inline Client/Frontend/Browser/BrowserViewController.swift]: Nit: We generally don't add parentheses around `if` conditions.

File: Client/Frontend/Home/HomePanelViewController.swift [hunk 4]
```diff
@@ -16,6 +16,9 @@ private struct HomePanelViewControllerUX {
 
 protocol HomePanelViewControllerDelegate: class {
     func homePanelViewController(homePanelViewController: HomePanelViewController, didSelectURL url: NSURL, visitType: VisitType)
+    
+    //adds function to check panel changes in home screen
```
Comment (thebnich) [inline Client/Frontend/Home/HomePanelViewController.swift]: You probably don't need a comment here either (`didSelectPanel` is pretty self-descriptive).

File: Client/Frontend/Browser/Browser.swift [hunk 5]
```diff
@@ -23,6 +23,7 @@ protocol BrowserDelegate {
 
 class Browser: NSObject {
     var webView: WKWebView? = nil
+    var previousHomePanel = 0
```
Comment (thebnich) [inline Client/Frontend/Browser/Browser.swift]: I think `lastHomePanel` would be a more appropriate name than `previousHomePanel`. "Previous" makes it sounds like we're on a current panel that's different from the one before, whereas "last" simply means the one that was last selected.

Comment (bkmunar) [inline Client/Frontend/Browser/Browser.swift]: finished

File: Client/Frontend/Browser/BrowserViewController.swift (hunk 1 above)
Comment (bkmunar) [inline Client/Frontend/Browser/BrowserViewController.swift]: finished

File: Client/Frontend/Browser/BrowserViewController.swift (hunk 2 above)
Comment (bkmunar) [inline Client/Frontend/Browser/BrowserViewController.swift]: finished

File: Client/Frontend/Browser/BrowserViewController.swift (hunk 3 above)
Comment (bkmunar) [inline Client/Frontend/Browser/BrowserViewController.swift]: finished

File: Client/Frontend/Home/HomePanelViewController.swift (hunk 4 above)
Comment (bkmunar) [inline Client/Frontend/Home/HomePanelViewController.swift]: finished

File: Client/Frontend/Home/HomePanelViewController.swift [hunk 6]
```diff
@@ -16,6 +16,8 @@ private struct HomePanelViewControllerUX {
 
 protocol HomePanelViewControllerDelegate: class {
     func homePanelViewController(homePanelViewController: HomePanelViewController, didSelectURL url: NSURL, visitType: VisitType)
+
```
Comment (thebnich) [inline Client/Frontend/Home/HomePanelViewController.swift]: Nit: no newline here

File: Client/Frontend/Browser/BrowserViewController.swift [hunk 7]
```diff
@@ -337,7 +337,9 @@ class BrowserViewController: UIViewController {
 
             addChildViewController(homePanelController!)
         }
-
+
```
Comment (thebnich) [inline Client/Frontend/Browser/BrowserViewController.swift]: Remove trailing whitespace here

File: Client/Frontend/Browser/BrowserViewController.swift [hunk 8]
```diff
@@ -337,7 +337,9 @@ class BrowserViewController: UIViewController {
 
             addChildViewController(homePanelController!)
         }
-
+        
+        homePanelController?.selectedButtonIndex = tabManager.selectedTab?.lastHomePanel
+
```
Comment (thebnich) [inline Client/Frontend/Browser/BrowserViewController.swift]: and here

File: Client/Frontend/Browser/BrowserViewController.swift [hunk 9]
```diff
@@ -891,6 +893,10 @@ extension BrowserViewController: HomePanelViewControllerDelegate {
     func homePanelViewController(homePanelViewController: HomePanelViewController, didSelectURL url: NSURL, visitType: VisitType) {
         finishEditingAndSubmit(url, visitType: visitType)
     }
+
```
Comment (thebnich) [inline Client/Frontend/Browser/BrowserViewController.swift]: and here
